shared/src/chain_state.rs

```diff
@@ -536,14 +532,16 @@ impl ChainState {
                     // InvalidHeader. They all represent invalid transactions
                     // that should just be discarded.
                     // OutOfOrder should only appear in BlockCellProvider
-                    UnresolvableError::Empty
-                    | UnresolvableError::UnspecifiedInputCell(_)
-                    | UnresolvableError::InvalidHeader(_)
-                    | UnresolvableError::OutOfOrder(_) => {
+                    OutPointError::MissingInputCellAndHeader
+                    | OutPointError::MissingInputCell(_)
+                    | OutPointError::ExclusiveInputCell(_)
+                    | OutPointError::UnknownHeader(_)
+                    | OutPointError::OutOfOrder(_) => {
```

Comment:
I am not sure this part, it looks like this PR includes more changes than it expected.